core-infrastructure/terraform/databases.tf

```diff
@@ -0,0 +1,233 @@
+  #checkov:skip=CKV2_AZURE_5:See ADO backlog AB#206493
+  server_security_alert_policy_id = azurerm_mssql_server_security_alert_policy.sql-security-alert-policy.id
+  storage_container_path          = "${azurerm_storage_account.sql-vulnerability-storage.primary_blob_endpoint}${azurerm_storage_container.sql-vulnerability-container.name}/"
+
+  recurring_scans {
+    enabled = true
+  }
+}
+
+resource "azurerm_storage_account" "sql-audit-storage" {
+  #checkov:skip=CKV_AZURE_43:False positive on storage account adhering to the naming rules
+  #checkov:skip=CKV2_AZURE_33:See ADO backlog AB#206389
+  #checkov:skip=CKV2_AZURE_1:See ADO backlog AB#206389
+  #checkov:skip=CKV_AZURE_59:See ADO backlog AB#206389
+  name                            = "${var.environment-prefix}audit"
+  location                        = azurerm_resource_group.resource-group.location
+  resource_group_name             = azurerm_resource_group.resource-group.name
+  account_tier                    = "Standard"
+  account_replication_type        = "GRS"
+  allow_nested_items_to_be_public = false
+  tags                            = local.common-tags
+  min_tls_version                 = "TLS1_2"
+  shared_access_key_enabled       = false
+
+  blob_properties {
+    delete_retention_policy {
+      days = 7
+    }
+  }
+
+  queue_properties {
+    logging {
+      delete                = true
+      read                  = true
+      write                 = true
+      version               = "1.0"
+      retention_policy_days = 10
+    }
+  }
+
+  network_rules {
+    default_action = "Deny"
+    bypass = [
+      "AzureServices"
+    ]
+  }
+}
+
+resource "azurerm_storage_account" "sql-threat-storage" {
+  #checkov:skip=CKV_AZURE_43:False positive on storage account adhering to the naming rules
+  #checkov:skip=CKV2_AZURE_33:See ADO backlog AB#206389
+  #checkov:skip=CKV2_AZURE_1:See ADO backlog AB#206389
+  #checkov:skip=CKV_AZURE_59:See ADO backlog AB#206389
+  name                            = "${var.environment-prefix}threat"
+  location                        = azurerm_resource_group.resource-group.location
+  resource_group_name             = azurerm_resource_group.resource-group.name
+  account_tier                    = "Standard"
+  account_replication_type        = "GRS"
+  allow_nested_items_to_be_public = false
+  tags                            = local.common-tags
+  min_tls_version                 = "TLS1_2"
+  shared_access_key_enabled       = false
+
+  blob_properties {
+    delete_retention_policy {
+      days = 7
+    }
+  }
+
+  queue_properties {
+    logging {
+      delete                = true
+      read                  = true
+      write                 = true
+      version               = "1.0"
+      retention_policy_days = 10
+    }
+  }
+
+  network_rules {
+    default_action = "Deny"
+    bypass = [
+      "AzureServices"
+    ]
+  }
+}
+
+resource "azurerm_storage_account" "sql-vulnerability-storage" {
```

Comment:
Can't the audit storage and vunerability storage all go to one core storage account for all databases

Reply:
Just checked container names on threat and audit. All three could go into a single storage account.